icefoganalytics/internal-data-portal
# Fix Unauthorized Error On Access to Production Site

Relates to:

- Introduced in https://github.com/icefoganalytics/internal-data-portal/pull/93 with commit https://github.com/icefoganalytics/internal-data-portal/commit/21c2559df10f993547ae0a078c6d66541b3dd8a4

# Context

Access to the production site fails with "Unauthorized" error.
Access to /_status route still succeeds showing that the app has booted succesfully.

![image](https://github.com/icefoganalytics/internal-data-portal/assets/23045206/0756cdfc-ae8e-4b9c-85e5-6487fd11e722)

# Implementation

:beetle: Re-scope base authentication action middlewares.

Avoid requiring auth for non-api and non-download prefixed routes. Move download from /datatets/:slug/download to /download/datasets/slug Move all the various controllers and services so they line up with the route.

# Testing Instructions

1. Run the test suite via `dev test` (or `dev test_api`)
2. Boot the app via:
    ```bash
    docker compose build --build-arg RELEASE_TAG=v2024.4.10.1 --build-arg GIT_COMMIT_HASH=97ffe6c183cad799fe78b2a3e33e7e34fe93ebfd

   docker compose up
    ```
3. Go to the app at http://localhost:8080.
4. While this will still fail due to the production Auth0 config not supporting redirect back to localhost, it will at least load the front-end and redirect to Auth0.
5. Stop the production build via `docker compose down`
6. Start a development build via `dev up`.
7. Check that you can log in to the app at http://localhost:8080.
8. Go to the /datasets page and pick a data with an integration or some data.
9. Go to the "Visualize" tab of the dataset.
10. Click the "Download to CSV" button, and check that the download works.
